src/domain/relayer/mod.rs

```diff
@@ -366,10 +366,26 @@ impl RelayerFactoryTrait for RelayerFactory {
                 Ok(NetworkRelayer::Solana(relayer))
             }
             NetworkType::Stellar => {
+                let network = match StellarNetwork::from_network_str(&relayer.network) {
+                    Ok(network) => network,
+                    Err(e) => return Err(RelayerError::NetworkConfiguration(e.to_string())),
+                };
+
+                let stellar_provider = StellarProvider::new(network.public_rpc_urls()[0])
```

Comment:
Let's have in mind active Seba's PR that improves this part for EVM and Solana. We should apply same approach here.

https://github.com/OpenZeppelin/openzeppelin-relayer/pull/182